### DISCUSSION TITLE
Error with collate function: nit() got an unexpected keyword argument '_base_cls'

### DISCUSSION THREAD

**Discussion by NavyaR13** [pyg-team/pytorch_geometric#8457] [General]
Traceback (most recent call last):
  File "/home/ubuntu/Desktop/project_cons/group_constpdb_to_pdb.py", line 37, in <module>
    train_loader, val_loader = Dataset_loader.create_traintestsplit(config,index_df)
  File "/home/ubuntu/Desktop/project_cons/data/dataset.py", line 105, in create_traintestsplit
    data_dict = collate_and_label_bags(data_dict, graph_type='pairs',label=None, ligand_id=None, exclude_keys=None, keep_dicts=False)
  File "/home/ubuntu/Desktop/project_cons/data/utils.py", line 55, in collate_and_label_bags
    print(collate(data_list=d1, cls=PairData,follow_batch=['atom_type_1', 'atom_type_2']))
  File "/home/ubuntu/anaconda3/envs/origami/lib/python3.9/site-packages/torch_geometric/data/collate.py", line 32, in collate
    out = cls(_base_cls=data_list[0].__class__)  # Dynamic inheritance.
TypeError: __init__() got an unexpected keyword argument '_base_cls'

Does anyone know how to fix this issue ? Any help is appreciated.

**Comment by NavyaR13** [comment]
Does it have something to do with versions ?

**Reply by rusty1s** [reply]
Yes, it looks like it. Do you know which versions were used to create the `Data` objects?

**Reply by NavyaR13** [reply]
Thank you for your response, the versions of packages I have used are here: 
PyTorch Version: 1.10.2
CUDA Version: 11.3
CUDNN Version: 8200
torch_geometric Version: 2.0.4
torch_cluster Version: 1.6.0
torch_sparse Version: 0.6.13

**Reply by rusty1s** [reply]
At best, I recommend to jump back to an earlier version of PyG, and save the data as a dictionary via `torch.save(data.to_dict()`, and then load it in later PyG versions via `Data.from_dict(torch.load(path))`. Luckily, this issue is no longer present in later PyG versions, but for older versions, this is the only possible workaround :(
